includes/scheduled-post-checker/scheduled-post-checker.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Newspack Scheduled Post Checker
+ * Checks to make sure posts haven't missed their schedule, and publishes them if needed.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Scheduled_Post_Checker;
+
+defined( 'ABSPATH' ) || exit;
+define( 'NEWSPACK_SCHEDULED_POST_CHECKER_CRON_HOOK', 'newspack_scheduled_post_checker' );
+
+/**
+ * Set up the checking.
+ */
+function nspc_init() {
+	register_deactivation_hook( __FILE__, 'nspc_deactivate' );
+	if ( ! wp_next_scheduled( NEWSPACK_SCHEDULED_POST_CHECKER_CRON_HOOK ) ) {
+		wp_schedule_event( time(), 'fivemins', NEWSPACK_SCHEDULED_POST_CHECKER_CRON_HOOK );
+	}
+}
+add_action( 'init', __NAMESPACE__ . '\nspc_init' );
+
+/**
+ * Clear the cron job when this plugin is deactivated.
+ */
+function nspc_deactivate() {
+	wp_clear_scheduled_hook( NEWSPACK_SCHEDULED_POST_CHECKER_CRON_HOOK );
+}
+
+/**
+ * Check to see if any posts have missed schedule, and try sending them live again if so.
+ */
+function nspc_run_check() {
+	$time = wp_date( 'Y-m-d H:i:s' );
+
+	$posts_with_missed_schedule = get_posts(
+		[
+			'post_status' => 'future',
+			'post_type'   => 'any',
+			'fields'      => 'ids',
+			'date_query'  => [
+				[
+					'before'    => $time,
+					'inclusive' => false,
+				],
+			],
+		]
+	);
+
+	if ( function_exists( 'spcl_log_event' ) ) {
```

Comment:
Where does `spcl_log_event` come from? A comment would be handy here.

Reply:
Great question, this comes directly from the original plugin. @claudiulodro, can you provide any context?

Reply:
All of the `spcl_` stuff is originally for a debug logger. We definitely don't want it in here now, since we know it works correctly and the `spcl_log_event ` function won't exist on any sites.

Reply:
Ah, thanks for the context! I've removed all references to that function in ca5ffb4.